crates/nargo_cli/src/cli/prove_cmd.rs

```diff
@@ -72,15 +76,32 @@ pub(crate) fn prove_with_path<B: Backend, P: AsRef<Path>>(
     check_proof: bool,
     compile_options: &CompileOptions,
 ) -> Result<Option<PathBuf>, CliError<B>> {
-    let preprocessed_program = match circuit_build_path {
-        Some(circuit_build_path) => read_program_from_file(circuit_build_path)?,
+    let common_reference_string = read_cached_common_reference_string();
```

Comment:
This logic with the following match statement and `write_cached_common_reference_string` looks to be repeated across multiple commands. Would we be able encapsulate this logic into its own method?

Reply:
These are explicitly separated to avoid unnecessary read/writes for contracts. You can see the merged PR for the discussion between @TomAFrench and me.